### PR TITLE
One Question Per Screen Updates

### PR DESCRIPTION
**Pull request by biyeun**
- handle submit in one question per screen mode
- know if index is end of form

related: https://github.com/dimagi/commcare-hq/pull/13441

@wpride @benrudolph 

cc @orangejenny

### CODE REVIEW

File: src/main/java/application/FormController.java [hunk 1]
```diff
@@ -363,8 +373,9 @@ public FormEntryResponseBean getPrevious(@RequestBody SessionRequestBean request
                     formSession.getFormEntryModel(),
                     formSession.getCurrentIndex());
             updateSession(formSession, serializableFormSession);
-            FormEntryResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryResponseBean.class);
+            FormEntryNavigationResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryNavigationResponseBean.class);
```
Comment (wpride) [inline src/main/java/application/FormController.java]: Method signature for this and getNext can be changed to FormEntryNavigationResponseBean

File: src/main/java/application/FormController.java [hunk 2]
```diff
@@ -341,7 +349,9 @@ public FormEntryResponseBean getNext(@RequestBody SessionRequestBean requestBean
...
+            responseBean.setIsAtLastIndex(formSession.getIsAtLastIndex());
```
Comment (wpride) [inline src/main/java/application/FormController.java]: I'd move this logic into the getCurrentJson method which should allow you to remove some of the logic out of FormSession as well

Comment (biyeun) [inline src/main/java/application/FormController.java]: Do you want me to change the `getCurrentJson` signature to accept additional params just for this use case? seems like more of a hassle than using it only where it's needed

Comment (wpride) [inline src/main/java/application/FormController.java]: Right but getCurrentJson does instantiate the object through the indexFromString method. Is also gettable through the getFormIndex() method of FormEntryModel. Writing that though I'm realizing we should definitely unify the way we're passing around the formIndex, though that's maybe out of scope for this PR.

**Comment by wpride** [comment]
@biyeun maybe for now add a getCurrentJson method to FormSession that makes the call to the util method, then adds the currentIndex and isLast variables itself. This helps keep the logic out of the controller (which I try to minimize since those classes tend towards hugeness)

**Comment by wpride** [comment]
(Accidental close)

**Comment by biyeun** [comment]
@wpride made those changes to `FormSession` as you suggested

**Pull request merged**